website/_dogfooding/clientModuleExample.ts

```diff
@@ -6,12 +6,33 @@
  */
 
 import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import type {Location} from 'history';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export function onRouteUpdate({location}: {location: Location}): void {
-  // console.log('onRouteUpdate', {location});
+export function onRouteUpdate({
+  location,
+  previousLocation,
+}: {
+  location: Location;
+  previousLocation: Location;
+}): void {
+  if (ExecutionEnvironment.canUseDOM) {
+    console.log(`onRouteUpdate (Fired before DOM repaints)
```

Comment:
it's not before dom repaints imho, it's more before React even starts to render

before dom repaints is useLayoutEffect (and we might not need this immediately)

Reply:
React commit -> DOM repaint so before React rerender is indeed before DOM repaint :D I see where you are coming from

Reply:
I understand it as "just before dom repaint, after React renders" (ie useLayoutEffect)

<img width="765" alt="image" src="https://user-images.githubusercontent.com/749374/155512633-59e0a962-efaa-493b-aa4b-c32bfe2ea194.png">